src/context/AuthContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { User } from '@supabase/supabase-js';
+
+import { ChildrenFC } from '../utils/type';
+import { getSupabaseClient } from '../api/supabase';
+
+type AuthContextType = {
+  user: User | null;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider: ChildrenFC = ({ children }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const supabase = getSupabaseClient();
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      const { data } = await supabase.auth.getUser();
+      setUser(data.user);
+    };
+
+    fetchUserData();
+  }, []);
+
+  useEffect(() => {
+    const {
+      data: { subscription },
+    } = supabase.auth.onAuthStateChange((_event, session) => {
+      setUser(session?.user ?? null);
+    });
+
+    return () => subscription.unsubscribe();
+  }, []);
+
+  return (
+    <AuthContext.Provider value={{ user }}>{children}</AuthContext.Provider>
```

Comment:
Passing ONLY the user if you have removed the useMemo is critical. Right now, you are passing an object that has a user object inside.

```suggestion
    <AuthContext.Provider value={user}>{children}</AuthContext.Provider>
```

I wonder if sonar was trying to warn you since we've set it up together and you ignored it :D, or if it stopped working.

<img width="808" alt="CleanShot 2025-01-08 at 20 05 05@2x" src="https://github.com/user-attachments/assets/ac9dedcc-9334-44b7-adfa-c781651aa596" />